test/gtest/bn_fwd_infer_ocl_hip.cpp

```diff
@@ -0,0 +1,274 @@
+        {"MIO_BN_GRP1", ylocalsize},
+        {"MIO_BN_GRP2", zlocalsize},
+        {"MIO_BN_GFX110X", (miopen::StartsWith(handle.GetDeviceName(), "gfx110") ? "1" : "0")},
+        {"MIO_BN_GFX103X", (miopen::StartsWith(handle.GetDeviceName(), "gfx103") ? "1" : "0")},
+    };
+
+    std::string kernel_file = "MIOpenBatchNormFwdInfer";
+    std::string kernel_name = "MIOpenBatchNormFwdInfer";
+
+    std::string params = use_hip ? build_params.GenerateFor(miopen::kbp::OpenCL{})
+                                 : build_params.GenerateFor(miopen::kbp::HIP{});
+
+    if(bn_mode == miopenBNSpatial)
+    {
+        if(use_hip)
+        {
+
+            kernel_file += "SpatialHIP.cpp";
+            kernel_name += "SpatialEstHIP";
+        }
+        else
+        {
+
+            kernel_file += "Spatial.cl";
+            kernel_name += "SpatialEst";
+        }
+    }
+    else
+    {
+
+        if(use_hip)
+        {
+
+            kernel_file += "PerActHIP.cpp";
+            kernel_name += "PerActivationEstHIP";
+        }
+        else
+        {
+
+            kernel_file += "PerAct.cl";
+            kernel_name += "PerActivationEst";
+        }
+    }
+
+    const std::vector<size_t> vgd{xgridsize, ygridsize, zgridsize};
+    const std::vector<size_t> vld{xlocalsize, ylocalsize, zlocalsize};
+
+    std::ostringstream ss;
+
+    ss << (use_hip ? "hip" : "ocl");
+    ss << "fp16" << static_cast<int>(xDesc.GetType() == miopenHalf);
+    ss << "fp32" << static_cast<int>(xDesc.GetType() == miopenFloat);
+    ss << "fp64" << static_cast<int>(xDesc.GetType() == miopenDouble);
+    ss << "fbf16" << static_cast<int>(xDesc.GetType() == miopenBFloat16);
+    ss << "mode" << bn_mode;
+    ss << "HWdims" << in_cstride;
+    ss << "C" << c;
+    ss << "layout" << xDesc.GetLayout_str();
+
+    std::string network_config = ss.str();
+
+    handle.AddKernel(kernel_name, network_config, kernel_file, kernel_name, vld, vgd, params)(
+        x,
+        y,
+        estimatedMean,
+        estimatedVariance,
+        bnScale,
+        bnBias,
+        epsilon,
+        n,
+        in_cstride,
+        in_nstride);
+}
+
+template <typename T>
+void BNTensorCompare(const tensor<T>& output,
+                     const tensor<T>& ref_out,
+                     const double threshold = std::numeric_limits<T>::epsilon())
+{
+    EXPECT_FALSE(miopen::range_zero(ref_out)) << "OCL data is all zeros";
+    EXPECT_FALSE(miopen::range_zero(output)) << "HIP data is all zeros";
+    EXPECT_FALSE(miopen::find_idx(output, miopen::not_finite) >= 0)
+        << "Non finite number found in the HIP data";
+    EXPECT_TRUE(miopen::range_distance(ref_out) == miopen::range_distance(output));
+    auto error = miopen::rms_range(ref_out, output);
+    EXPECT_FALSE(miopen::find_idx(ref_out, miopen::not_finite) >= 0)
+        << "Non finite number found in the OCL data";
+
+    EXPECT_TRUE(error < threshold)
+        << "Error beyond tolerance Error:" << error << ",  Threshold: " << threshold;
```

Comment:
```suggestion
    EXPECT_LT(error, threshold);
```
Also check the possible gtest assertion predicates:
https://google.github.io/googletest/reference/assertions.html